src/iosched/unified.c

```diff
@@ -450,22 +496,26 @@ int unified_close(struct dentry *d, bool flush, void *iosched_handle)
 {
 	int write_error, ret = 0;
 	struct unified_data *priv = iosched_handle;
+	struct dentry_priv *dpr;
 
 	CHECK_ARG_NULL(d, -LTFS_NULL_ARG);
 	CHECK_ARG_NULL(iosched_handle, -LTFS_NULL_ARG);
 	ltfs_profiler_add_entry(priv->profiler, &priv->proflock, IOSCHED_REQ_ENTER(REQ_IOS_CLOSE));
 
 	acquireread_mrsw(&priv->lock);
 	ltfs_mutex_lock(&d->iosched_lock);
+	ret = _unified_get_dentry_priv(d, &dpr, priv);
```

Comment:
Here, the value of 'ret' is not checked (and it's potentially overwritten by the branch condition below)